### DISCUSSION TITLE
Using search params for controlled filter (app router)

### DISCUSSION THREAD

**Discussion by tar-aldev** [vercel/next.js#60791] [Help]
### Summary

I am facing a problem with app router and search params.
Due to requirements, I can't use uncontrolled inputs, checkboxes, .... as shown in tutorial https://nextjs.org/learn/dashboard-app/adding-search-and-pagination#3-keeping-the-url-and-input-in-sync.
The reason I can't do it, is because I can add filters to url in one place (sidebar) and then remove filters from another place on the page.

The problem is that, when I change filter and network speed is slow, the following happens:
1. I click on checkbox to add filter to URL
2. next.js requests RSC payload
3. url is updated
4. checkbox reads updated url and is updated to checked state

With slow network, the UI becomes unresponsive.
Is there anything I can do? Before, I guess I could use shallow routing and fetch subsequent data on the client, but with next.js 13 it doesn't seem to be possible.

I saw an option with parallel routes, when you have filters as one route and content as another route. However, I a not sure if that would help, because next.js will still request full page RSC so filters will become unresponsive...

**Comment by hedysnike** [comment]
VERCEL PLEASE PROVIDE FUEL FOR JET (aka shallow routing for app router)

**Reply by hedysnike** [reply]
WE NEED IT MORE THEN WATER OR EVEN OXYGEN
